### ISSUE TITLE
#or and #ref tags don't combine as expected in "large" maps

### ISSUE THREAD

**Issue by mchughs** [juxt/aero#82]
On aero 1.1.4 the #or and #ref tags do not combined as expected when the maps contain enough data.

The following config map
```clojure
 {:x "foo"
  :y #or [#ref [:x] "bar"]}
```
will properly evaluate to
```clojure
 {:x "foo"
  :y "foo"}
```

However the config map
```clojure
 {:x "foo"
  :y #or [#ref [:x] "bar"]
  :junk0 "0"
  :junk1 "1"
  :junk2 "2"
  :junk3 "3"
  :junk4 "4"
  :junk5 "5"
  :junk6 "6"
  :junk7 "7"
  :junk8 "8"}
```
improperly evaluates to
```clojure
 {:x "foo"
  :y nil
  :junk0 "0"
  :junk1 "1"
  :junk2 "2"
  :junk3 "3"
  :junk4 "4"
  :junk5 "5"
  :junk6 "6"
  :junk7 "7"
  :junk8 "8"}
```

**Comment by SevereOverfl0w** [comment]
How strange. I don't have time to dig into this deeply at the moment, I would welcome a pull request that fixed this of course.

**Comment by malcolmsparks** [comment]
I wonder if this is one of those array-map/hash-map thresholds...

**Comment by SevereOverfl0w** [comment]
Was doing some benchmarking and curiosity got the better of me. There's a bug in `or`. It isn't returning that it is incomplete when one of it's children are incomplete.

**Comment by mchughs** [comment]
Fantastic. Thank you so much.

**Comment by dharrigan** [comment]
This was affecting me too. Will this be ported to v3?

**Comment by SevereOverfl0w** [comment]
You can try #83 as a fix, but we aren't certain about merging yet. I have some more thinking to do.

**Comment by dharrigan** [comment]
I'm currently on the V3 branch as it contains fixes that are important to me, i.e., handing a `#ref` inside a function etc...

**Comment by SevereOverfl0w** [comment]
That's been merged and released as 1.1.4 (aka master)

**Comment by SevereOverfl0w** [comment]
@dharrigan #83 is much more likely to be merged now, I think I've now ironed things out.  Your testing would be very appreciated to ensure there's no other side effects.
